Add Genius as a lyric source

sadatay opened the pull request:
Makes use of the Genius API and genius-api.com, a separate resource that
provides readable lyrics.

Fixes #1626.

This is my first contribution to beets so please let me know if anything is amiss.  I'd like to wrap some testing around this but I'm not particularly familiar with python testing frameworks, so that may take a little while longer.

Requires a new config option ```genius_API_key```, and ```requests[security]``` to quell InsecureResponseWarnings.

sampsyo commented on beetsplug/lyrics.py:
```diff
@@ -459,12 +508,14 @@ def __init__(self):
         })
         self.config['google_API_key'].redact = True
         self.config['google_engine_ID'].redact = True
+        self.config['genius_API_key'].redact = True
```
Can we use an all-lower-case config name, `genius_api_key`? I know this doesn't match the other options just above, but it does comport better with the rest of beets.

sampsyo commented:
This looks awesome! It will be great to have. A couple of outstanding notes:

* We should mention the new backend (and its configuration) in the documentation for the plugin.
* Should we apply for an official beets API key, as we do for some services, or just leave it up to the user?
* It looks like Travis is reporting a few minor style issues.

sadatay commented:
I've added some documentation and fixed the style checks.  Registering the app to genius might be a good idea since we're only doing reads and they aren't user specific, I'll look into that this evening.  Is that something I should register or would it be more appropriate for the owner of the app or plugin?

sampsyo commented:
Awesome!

Please do go for it. We're an open source hive-mind, so you're perfectly authorized to get the key. :smiley:

sadatay commented on beetsplug/lyrics.py:
```diff
@@ -459,12 +524,16 @@ def __init__(self):
             'google_engine_ID': u'009217259823014548361:lndtuqkycfu',
+            'genius_api_key':
+                "Ryq93pUGm8bM6eUWwD_M3NOFFDAtp2yEE7W"
+                "76V-uFL5jks5dNvcGCdarqFjDhP9c",
```
this is the api auth key for the app, which is now registered with genius.  let me know if there's a better place to put this; i'm not crazy about inlining it but since the lyrics plugin is not enabled by default it wouldn't make any sense to put it in ```config_default.yaml```.

sampsyo replied:
Seems great. Yeah, it's a little strange, but this is indeed the idiomatic way to specify config defaults for plugins.

sadatay commented on beetsplug/lyrics.py:
```diff
@@ -459,12 +524,16 @@ def __init__(self):
         })
         self.config['google_API_key'].redact = True
         self.config['google_engine_ID'].redact = True
+        self.config['genius_api_key'].redact = True
```
is this line still necessary now that it's no longer a config option?  i'm not quite sure what is going on here.

sampsyo replied:
Let's leave it, just to avoid accidentally exposing anyone's private API key if they opt to use a custom one.

sampsyo commented:
This looks great! Basically ready to merge. Two outstanding things:

* Would you mind adding a brief note to the changelog about the addition?
* Out of curiosity, why does this need `requests[security]`? Any chance there's a way around it, since beets users probably already have requests but not the security extra?

sadatay commented:
Heya,

For the first bullet point, how does the versioning work?  Should I add a note about this under 1.3.15?  I'm going to go ahead and push that out, let me know if it's incorrect.

Also, after a bit more research on the issue which prompted me to install requests[security], I'm pretty sure it's not actually necessary at all.  I'm currently running python 2.7.6 and according to this stackoverflow thread 

>Requests 2.6 introduced this warning for users of python prior to 2.7.9 with only stock SSL modules available.

So I think this was just an issue with my system being configured poorly.  I'll remove the dependency from the documentation.

sampsyo commented:
Yes, under the latest version (1.3.15 at the moment) should do perfectly.

And awesome that the dependency situation can be simplified.

Thank you again! :sparkles:

sampsyo commented:
All merged. Thanks once again.
